### PR TITLE
Add a new ten site example domain for testing

### LINKED ISSUE #622: Add new 10 site test Domain to testing folder in ADRIA
A new ten site test domain has been developed with properties which allow testing and exploration of location selection in ADRIA. This test set is required for the new test suite being developed (#621 )but may require changes in the current set of tests which use the old test domain in ADRIA.

### PR DESCRIPTION
**Pull request by Rosejoycrocker**
Closes #622

All tests running.

### CODE REVIEW

File: examples/Example_domain/datapackage.json
Comment (ConnectedSystems) [inline examples/Example_domain/datapackage.json:2]: The standard for JSON is to indent using 2 spaces, not four.

File: examples/Example_domain/connectivity/2000/example_conn.csv
Comment (ConnectedSystems) [inline examples/Example_domain/connectivity/2000/example_conn.csv:1]: Missing metadata?

File: examples/Example_domain/datapackage.json
Comment (ConnectedSystems) [inline examples/Example_domain/datapackage.json:46]: Hmm, old legacy description - please update.

File: examples/Example_domain/datapackage.json
Comment (ConnectedSystems) [inline examples/Example_domain/datapackage.json:85]: There should be a newline at end of file.

**Comment by ConnectedSystems** [comment]
There are duplicate datasets in this PR (one in the `Example_domain` and a few separate files in the `test` directory.
Okay to leave this as is but in future I think if we're doing these kinds of replacements, we might as well get rid of duplicates and simply update tests to use the new data.

The `Example_domain` would then sit in the `test` folder and be called `Test_domain`. This would help resolve #578 .

Running tests to confirm they're passing now.

File: examples/Example_domain/datapackage.json
Comment (ConnectedSystems) [reply examples/Example_domain/datapackage.json:2]: Also, shouldn't this be updated to `Test_example` if they are intended for testing?

**Comment by ConnectedSystems** [comment]
@Zapiano this PR fails tests due to `LoadError: type Rule has no field clause`.

Stacktrace points to this line:

https://github.com/open-AIMS/ADRIA.jl/blob/c5ebc5995a0957417b2ada7faccfdd7c52ff1290/src/analysis/rule_extraction.jl#L31-L32

Full trace below

```julia
ERROR: LoadError: type Rule has no field clause
Stacktrace:
  [1] getproperty
    @ .\Base.jl:37 [inlined]
  [2] _condition(rules::SIRUS.StableRules{Int64}, index::Int64)
    @ ADRIA.analysis C:\Users\tiwanaga\projects\ADRIA.jl\src\analysis\rule_extraction.jl:50
  [3] (::ADRIA.analysis.var"#31#32"{SIRUS.StableRules{Int64}})(i::Int64)
    @ ADRIA.analysis .\none:0
  [4] iterate
    @ .\generator.jl:47 [inlined]
  [5] collect
    @ .\array.jl:782 [inlined]
  [6] rules
    @ C:\Users\tiwanaga\projects\ADRIA.jl\src\analysis\rule_extraction.jl:31 [inlined]
  [7] cluster_rules(clusters::Vector{Int64}, X::DataFrame, max_rules::Int64; n_trees::Int64, seed::Int64)
    @ ADRIA.analysis C:\Users\tiwanaga\projects\ADRIA.jl\src\analysis\rule_extraction.jl:145
  [8] cluster_rules
    @ C:\Users\tiwanaga\projects\ADRIA.jl\src\analysis\rule_extraction.jl:125 [inlined]
  [9] test_rs_w_fig()
    @ Main C:\Users\tiwanaga\projects\ADRIA.jl\test\runtests.jl:280
 [10] top-level scope
    @ C:\Users\tiwanaga\projects\ADRIA.jl\test\runtests.jl:335
 [11] include(fname::String)
    @ Base.MainInclude .\client.jl:478
 [12] top-level scope
    @ none:6
in expression starting at C:\Users\tiwanaga\projects\ADRIA.jl\test\runtests.jl:335
ERROR: Package ADRIA errored during testing
```

**Comment by Rosejoycrocker** [comment]
> @Zapiano this PR fails tests due to `LoadError: type Rule has no field clause`.
> 
> Stacktrace points to this line:
> 
> https://github.com/open-AIMS/ADRIA.jl/blob/c5ebc5995a0957417b2ada7faccfdd7c52ff1290/src/analysis/rule_extraction.jl#L31-L32
> 
> Full trace below
> 
> ```julia
> ERROR: LoadError: type Rule has no field clause
> Stacktrace:
>   [1] getproperty
>     @ .\Base.jl:37 [inlined]
>   [2] _condition(rules::SIRUS.StableRules{Int64}, index::Int64)
>     @ ADRIA.analysis C:\Users\tiwanaga\projects\ADRIA.jl\src\analysis\rule_extraction.jl:50
>   [3] (::ADRIA.analysis.var"#31#32"{SIRUS.StableRules{Int64}})(i::Int64)
>     @ ADRIA.analysis .\none:0
>   [4] iterate
>     @ .\generator.jl:47 [inlined]
>   [5] collect
>     @ .\array.jl:782 [inlined]
>   [6] rules
>     @ C:\Users\tiwanaga\projects\ADRIA.jl\src\analysis\rule_extraction.jl:31 [inlined]
>   [7] cluster_rules(clusters::Vector{Int64}, X::DataFrame, max_rules::Int64; n_trees::Int64, seed::Int64)
>     @ ADRIA.analysis C:\Users\tiwanaga\projects\ADRIA.jl\src\analysis\rule_extraction.jl:145
>   [8] cluster_rules
>     @ C:\Users\tiwanaga\projects\ADRIA.jl\src\analysis\rule_extraction.jl:125 [inlined]
>   [9] test_rs_w_fig()
>     @ Main C:\Users\tiwanaga\projects\ADRIA.jl\test\runtests.jl:280
>  [10] top-level scope
>     @ C:\Users\tiwanaga\projects\ADRIA.jl\test\runtests.jl:335
>  [11] include(fname::String)
>     @ Base.MainInclude .\client.jl:478
>  [12] top-level scope
>     @ none:6
> in expression starting at C:\Users\tiwanaga\projects\ADRIA.jl\test\runtests.jl:335
> ERROR: Package ADRIA errored during testing
> ```

This error should be fixed since the most recent rebase, this is in the most updated version of `add-new-testset`?

**Comment by ConnectedSystems** [comment]
> This error should be fixed since the most recent rebase, this is in the most updated version of add-new-testset?

The branch you mention seems to be a new one you just created, based on an older commit?

image
image


You should see that both branches are based on `Main` anyway so the error is still there.

**Comment by ConnectedSystems** [comment]
Hang on one tick, suspect my test environment didn't fully update. Will check again soon.

**Comment by ConnectedSystems** [comment]
@Rosejoycrocker tests pass on `Main` but fail in `add-new-testset` due to missing package `IterTools`

Looks like you added it but did not add as dependency.

https://github.com/open-AIMS/ADRIA.jl/blob/c5ebc5995a0957417b2ada7faccfdd7c52ff1290/test/site_selection.jl#L2

If this dependency is for tests only, you should specify it here: https://github.com/open-AIMS/ADRIA.jl/blob/c5ebc5995a0957417b2ada7faccfdd7c52ff1290/Project.toml#L136-L137

**Comment by Rosejoycrocker** [comment]
> @Rosejoycrocker tests pass on `Main` but fail on `add-new-testset` due to missing package `IterTools`
> 
> Looks like you added it but did not add as dependency.
> 
> https://github.com/open-AIMS/ADRIA.jl/blob/c5ebc5995a0957417b2ada7faccfdd7c52ff1290/test/site_selection.jl#L2
> 
> If this dependency is for tests only, you should specify it here:
> 
> https://github.com/open-AIMS/ADRIA.jl/blob/c5ebc5995a0957417b2ada7faccfdd7c52ff1290/Project.toml#L136-L137

Thanks Takuya, it is just for tests. Do I just add it to the Project.toml manually?

**Comment by ConnectedSystems** [comment]
> > @Rosejoycrocker tests pass on `Main` but fail on `add-new-testset` due to missing package `IterTools`
> > Looks like you added it but did not add as dependency.
> > https://github.com/open-AIMS/ADRIA.jl/blob/c5ebc5995a0957417b2ada7faccfdd7c52ff1290/test/site_selection.jl#L2
> > 
> > If this dependency is for tests only, you should specify it here:
> > https://github.com/open-AIMS/ADRIA.jl/blob/c5ebc5995a0957417b2ada7faccfdd7c52ff1290/Project.toml#L136-L137
> 
> Thanks Takuya, it is just for tests. Do I just add it to the Project.toml manually?

Are you still waiting on a response to this or are you good?

File: examples/Example_domain/datapackage.json
Comment (ConnectedSystems) [reply examples/Example_domain/datapackage.json]: This is still formatted incorrectly.

File: examples/Example_domain/datapackage.json
Comment (ConnectedSystems) [inline examples/Example_domain/datapackage.json:2]: This file is still formatted incorrectly.

Two-space indentation please.

File: examples/Test_domain/connectivity/2000/example_conn.csv
Comment (ConnectedSystems) [inline examples/Test_domain/connectivity/2000/example_conn.csv:3]: Please don't edit these with Excel, it automatically messes up the formatting.

Delete the additional columns at end of line.

File: examples/Test_domain/datapackage.json
Comment (ConnectedSystems) [inline examples/Test_domain/datapackage.json:12]: Okay, I think maybe we're not on the same page.

There seems to be two identical data packages now. One called `Test_domain` and another called `Example_domain`.
I expected only one data package called `Test_domain`.

In a later PR, we would move this to the `test` directory and get rid of the `examples` directory all together. 

As discussed earlier, the `examples` are now shown in the documentation so we have no need for it any more.

**Comment by Rosejoycrocker** [comment]
> > > @Rosejoycrocker tests pass on `Main` but fail on `add-new-testset` due to missing package `IterTools`
> > > Looks like you added it but did not add as dependency.
> > > https://github.com/open-AIMS/ADRIA.jl/blob/c5ebc5995a0957417b2ada7faccfdd7c52ff1290/test/site_selection.jl#L2
> > > 
> > > If this dependency is for tests only, you should specify it here:
> > > https://github.com/open-AIMS/ADRIA.jl/blob/c5ebc5995a0957417b2ada7faccfdd7c52ff1290/Project.toml#L136-L137
> > 
> > 
> > Thanks Takuya, it is just for tests. Do I just add it to the Project.toml manually?
> 
> Are you still waiting on a response to this or are you good?

Still waiting, I have added it manually but it throws an error saying that the dependency is not in deps, weakdeps or extras. I'm guessing I should add it to weakdeps but not sure how

**Comment by Rosejoycrocker** [comment]
> There are duplicate datasets in this PR (one in the `Example_domain` and a few separate files in the `test` directory). Okay to leave this as is but in future I think if we're doing these kinds of replacements, we might as well get rid of duplicates and simply update tests to use the new data.
> 
> The `Example_domain` would then sit in the `test` folder and be called `Test_domain`. This would help resolve #578 .
> 
> Running tests to confirm they're passing now.

There should be one dataset- the `Example_domain` folder seems to be a remnant from me changing the names of files  (it only contains the datapackage json), but I've deleted it now

File: examples/Example_domain/datapackage.json
Comment (Rosejoycrocker) [reply examples/Example_domain/datapackage.json]: Seems my vs code keeps resetting the indentation, I'll see wha I need to change in the settings

**Comment by ConnectedSystems** [comment]
> Still waiting, I have added it manually but it throws an error saying that the dependency is not in deps, weakdeps or extras. I'm guessing I should add it to weakdeps but not sure how

We shouldn't let something this small block an entire PR.
Google each of those and learn what they are and take the appropriate course of action please.

**Comment by Rosejoycrocker** [comment]
> > Still waiting, I have added it manually but it throws an error saying that the dependency is not in deps, weakdeps or extras. I'm guessing I should add it to weakdeps but not sure how
> 
> We shouldn't let something this small block an entire PR. Google each of those and learn what they are and take the appropriate course of action please.

I was actually in the process of working it out when you commented on it, it got left for 2 days because of the workshops

**Comment by Rosejoycrocker** [comment]
I think I've addressed your comments now @ConnectedSystems

File: examples/Test_domain/connectivity/2000/example_conn.csv
Comment (ConnectedSystems) [inline examples/Test_domain/connectivity/2000/example_conn.csv:1]: You missed one

Comment (Rosejoycrocker) [reply examples/Test_domain/connectivity/2000/example_conn.csv]: All fixed now, I thought it was part of the sentence

Comment (Rosejoycrocker) [reply examples/Test_domain/connectivity/2000/example_conn.csv]: Tests all passing

**Comment by ConnectedSystems** [comment]
Approving, assuming the connectivity matrix will be revisited later.

**Review by ConnectedSystems — APPROVED** [review]
Approving, assuming the connectivity matrix will be revisited later, and that the Test domain will be moved to the test directory at some point.

**Pull request merged**
